src/ddprof_worker.cc

```diff
@@ -607,7 +662,7 @@ DDRes ddprof_worker_cycle(DDProfContext *ctx, int64_t now,
   }
   unwind_cycle(ctx->worker_ctx.us);
 
-  clear_unvisted_pids(ctx->worker_ctx);
+  DDRES_CHECK_FWD(clear_unvisted_pids(ctx));
```

Comment:
It might make more sense to check pids before sending profiles, that way processes that died in this cycle won't be reported on the next cycle.

Reply:
Good point, this is better indeed. So
- Clear anything that died. Report it at the time of clean up
- Do the cycle after
This ensures we don't see the information in the next cycle.